client/components/sites-dropdown/index.jsx

```diff
@@ -34,39 +36,35 @@ export default class SitesDropdown extends PureComponent {
 		isPlaceholder: false
 	}
 
-	constructor( props ) {
-		super( props );
-
-		this.selectSite = this.selectSite.bind( this );
-		this.toggleOpen = this.toggleOpen.bind( this );
-		this.onClose = this.onClose.bind( this );
-
-		const selectedSite = props.selectedSiteId
-			? sites.getSite( props.selectedSiteId )
-			: sites.getPrimary();
-
-		this.state = {
-			selectedSiteSlug: selectedSite && selectedSite.slug
-		};
+	state = {
+		selectedSiteId: this.props.selectedOrPrimarySiteId,
 	}
 
-	getSelectedSite() {
-		return sites.getSite( this.state.selectedSiteSlug );
-	}
+	selectSite = ( siteOrsiteID ) => {
+		const ID = Number.isInteger( siteOrsiteID )
+			? siteOrsiteID
+			: get( siteOrsiteID, 'ID', null );
```

Comment:
Oh, I missed 1841c4dd32f88e7ff696b797a4036fe5f85f6c39; need to look at that. (I still stand by avoiding extraneous degrees of freedom tho :slightly_smiling_face: )